File: Easy/longest_common_prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not len(strs): return ""
        
        strs.sort(key=len)
        prf = strs[0]

        for w in strs[1:]:
            prf2 = ''
            for i, c in enumerate(w[:len(prf)]):
                if c == prf[i]: prf2 += c
                else: break
 
            prf = prf2 
            
        return prf

File: Easy/Swift/longest_common_prefix.swift
class Solution {
    func longestCommonPrefix(_ strs: [String]) -> String {
        guard strs.count > 0 else { return "" }
        
        var commonPrefix = strs.first!
        
        for str in strs[1...] {
            while !str.hasPrefix(commonPrefix) {
                commonPrefix = String(commonPrefix.dropLast())
            }
        }

        return commonPrefix
    }
}
